plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -54,58 +54,98 @@ function perflab_query_plugin_info( string $plugin_slug ) {
+			$plugins[ $standalone_plugin ] = $plugins[ $plugin_slug ];
+		}
 
-	// Start processing the plugins using a queue-based approach.
-	while ( count( $plugin_queue ) > 0 ) { // phpcs:ignore Squiz.PHP.DisallowSizeFunctionsInLoops.Found
-		$current_plugin_slug = array_shift( $plugin_queue );
+		$has_errors = true;
+	} else {
+		$plugin_queue = perflab_get_standalone_plugins();
 
-		if ( isset( $plugins[ $current_plugin_slug ] ) ) {
-			continue;
-		}
+		// Index the plugins from the API response by their slug for efficient lookup.
+		$all_performance_plugins = array_column( $response->plugins, null, 'slug' );
 
-		if ( ! isset( $all_performance_plugins[ $current_plugin_slug ] ) ) {
-			return new WP_Error(
-				'plugin_not_found',
-				__( 'Plugin not found in WordPress.org API response.', 'performance-lab' )
-			);
+		// Start processing the plugins using a queue-based approach.
+		while ( count( $plugin_queue ) > 0 ) { // phpcs:ignore Squiz.PHP.DisallowSizeFunctionsInLoops.Found
+			$current_plugin_slug = array_shift( $plugin_queue );
+
+			if ( isset( $plugins[ $current_plugin_slug ] ) ) {
+				continue;
+			}
+
+			if ( ! isset( $all_performance_plugins[ $current_plugin_slug ] ) ) {
+				// Cache the fact that the plugin was not found.
+				$plugins[ $current_plugin_slug ] = array(
+					'error' => array(
+						'code'    => 'plugin_not_found',
+						'message' => __( 'Plugin not found in API response.', 'performance-lab' ),
+					),
+				);
+
+				$has_errors = true;
+				continue;
+			}
+
+			$plugin_data                     = $all_performance_plugins[ $current_plugin_slug ];
+			$plugins[ $current_plugin_slug ] = wp_array_slice_assoc( $plugin_data, $fields );
+
+			// Enqueue the required plugins slug by adding it to the queue.
+			if ( isset( $plugin_data['requires_plugins'] ) && is_array( $plugin_data['requires_plugins'] ) ) {
+				$plugin_queue = array_merge( $plugin_queue, $plugin_data['requires_plugins'] );
+			}
 		}
 
-		$plugin_data                     = $all_performance_plugins[ $current_plugin_slug ];
-		$plugins[ $current_plugin_slug ] = wp_array_slice_assoc( $plugin_data, $fields );
+		if ( ! isset( $plugins[ $plugin_slug ] ) ) {
+			// Cache the fact that the plugin was not found.
+			$plugins[ $plugin_slug ] = array(
+				'error' => array(
+					'code'    => 'plugin_not_found',
+					'message' => __( 'The requested plugin is not part of Performance Lab plugins.', 'performance-lab' ),
+				),
+			);
 
-		// Enqueue the required plugins slug by adding it to the queue.
-		if ( isset( $plugin_data['requires_plugins'] ) && is_array( $plugin_data['requires_plugins'] ) ) {
-			$plugin_queue = array_merge( $plugin_queue, $plugin_data['requires_plugins'] );
+			$has_errors = true;
 		}
 	}
 
-	set_transient( $transient_key, $plugins, HOUR_IN_SECONDS );
+	if ( $has_errors ) {
+		set_transient( $transient_key, $plugins, MINUTE_IN_SECONDS );
+	} else {
+		set_transient( $transient_key, $plugins, HOUR_IN_SECONDS );
+	}
```

Comment:
To reduce code duplication:

```suggestion
	set_transient( $transient_key, $plugins, $has_errors ? MINUTE_IN_SECONDS : HOUR_IN_SECONDS );
```